### DISCUSSION TITLE
BIPES, CircuitPython e ESP32S2 com Franzininho WiFi

### DISCUSSION THREAD

**Discussion by rafaelaroca** [BIPES/BIPES#101] [Português] [answered]
Mais informações sobre o uso do BIPES com o Franzininho WiFi:

https://rafaelaroca.wordpress.com/2022/01/03/post-in-portuguese-franzininho-wifi-e-bipes/

**Comment by libertos** [comment]
Hi, I would ask you about an issue. I have this message, how can solve it?  thank you
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: no module named 'machine'

**Reply by rafaelaroca** [reply]
Hello. Strange. Did you select Franzininho WiFi / ESP32S2 on the board list? 

import machine is used when ESP32 or ESP8266 is selected. For ESP32S2 other imports are generated.

best regards
Rafael

**Reply by libertos** [reply]
Hi, thank you for your response.
I selected Franzininho WiFi / ESP32S2 and the problem remain.
![issue](https://user-images.githubusercontent.com/26169727/149615186-1649f180-372c-490a-8e78-3b85b4c7d18a.jpg)

**Reply by KK4HFJ** [reply]
I get this error when Raspberry Pi Pico is selected and I try to blink the onboard LED.

**Reply by rafaelaroca** [reply]
> I get this error when Raspberry Pi Pico is selected and I try to blink the onboard LED.

Hello, could you please give more details? This post is based on MicroPython + ESP32-S2. You mentioned that you are using Raspberry Pi Pico. Are you using Pico with MicroPython or CircuitPython.

I just tried the link and the code generated (ESP32-S2 / CircuitPython) is ok:

import board
from digitalio import DigitalInOut, Direction, Pull
import time

try:
	gpio4.deinit()
except:
	pass
gpio4=DigitalInOut(board.IO4)
gpio4.direction = Direction.OUTPUT


while True:
  gpio4.value=True
  time.sleep(0.5)
  gpio4.value=False
  time.sleep(0.5)


Here: https://bipes.net.br/beta2/ui/?lang=pt-br#an9f6p

[]s

**Reply by KK4HFJ** [reply]
it generated the following from your blink sample:

```
from machine import Pin
import time

def gpio_set(pin,value):
  if value >= 1:
    Pin(pin, Pin.OUT).on()
  else:
    Pin(pin, Pin.OUT).off()


while True:
  gpio_set(25, True)
  time.sleep(2)
  gpio_set(25, False)
  time.sleep(2)
```

**Reply by KK4HFJ** [reply]
console says:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: no module named 'machine'

**Comment by rafaelaroca** [comment, marked as answer]
HMMM! The problem is with the block for the DHT11 sensor. We have not implemented the DHT11 sensor block for ESP32S2, so this is why you are getting the error.

If possible, I ask/suggest you build your own block for the DHT11 by creating a function that runs Python code. That is relatively easy.

For example:

http://bipes.net.br/beta2/ui/#2wjiqv

This is an example of running Linux commands. You should change it to run the DHT11 related commands on the ESP32S2. Once you test it, if you can share it with us, we can implement it as a block on the platform.

Regards.
